15.1.0 RC1: crash when opening media overview

### Actual behaviour

crash because dateUtils has not been initialized


```
2022-12-19 13:59:39.222 31093-31093 AndroidRuntime          com.nextcloud.talk2                  E  FATAL EXCEPTION: main
                                                                                                    Process: com.nextcloud.talk2, PID: 31093
                                                                                                    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.nextcloud.talk2/com.nextcloud.talk.shareditems.activities.SharedItemsActivity}: kotlin.UninitializedPropertyAccessException: lateinit property dateUtils has not been initialized
                                                                                                    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3311)
                                                                                                    	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3460)
                                                                                                    	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2047)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:107)
                                                                                                    	at android.os.Looper.loop(Looper.java:224)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7590)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
                                                                                                    Caused by: kotlin.UninitializedPropertyAccessException: lateinit property dateUtils has not been initialized
                                                                                                    	at com.nextcloud.talk.dagger.modules.RepositoryModule.getDateUtils(RepositoryModule.kt:57)
                                                                                                    	at com.nextcloud.talk.dagger.modules.RepositoryModule.provideSharedItemsRepository(RepositoryModule.kt:66)
                                                                                                    	at com.nextcloud.talk.dagger.modules.RepositoryModule_ProvideSharedItemsRepositoryFactory.provideSharedItemsRepository(RepositoryModule_ProvideSharedItemsRepositoryFactory.java:47)
                                                                                                    	at com.nextcloud.talk.dagger.modules.RepositoryModule_ProvideSharedItemsRepositoryFactory.get(RepositoryModule_ProvideSharedItemsRepositoryFactory.java:37)
                                                                                                    	at com.nextcloud.talk.dagger.modules.RepositoryModule_ProvideSharedItemsRepositoryFactory.get(RepositoryModule_ProvideSharedItemsRepositoryFactory.java:13)
                                                                                                    	at com.nextcloud.talk.shareditems.viewmodels.SharedItemsViewModel_Factory.get(SharedItemsViewModel_Factory.java:31)
                                                                                                    	at com.nextcloud.talk.shareditems.viewmodels.SharedItemsViewModel_Factory.get(SharedItemsViewModel_Factory.java:11)
                                                                                                    	at com.nextcloud.talk.dagger.modules.ViewModelFactory.create(ViewModelModule.kt:44)
                                                                                                    	at androidx.lifecycle.ViewModelProvider$Factory.create(ViewModelProvider.kt:83)
                                                                                                    	at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.kt:187)
                                                                                                    	at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.kt:153)
                                                                                                    	at com.nextcloud.talk.shareditems.activities.SharedItemsActivity.onCreate(SharedItemsActivity.kt:88)
                                                                                                    	at android.app.Activity.performCreate(Activity.java:7893)
                                                                                                    	at android.app.Activity.performCreate(Activity.java:7880)
                                                                                                    	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1307)
                                                                                                    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3286)
                                                                                                    	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3460) 
                                                                                                    	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2047) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:107) 
                                                                                                    	at android.os.Looper.loop(Looper.java:224) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7590) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950) 
```


### Nextcloud Talk app version

15.1.0 RC1

--- a/app/src/main/java/com/nextcloud/talk/dagger/modules/RepositoryModule.kt
+++ b/app/src/main/java/com/nextcloud/talk/dagger/modules/RepositoryModule.kt
@@ -48,13 +48,9 @@ import com.nextcloud.talk.utils.database.user.CurrentUserProviderNew
 import dagger.Module
 import dagger.Provides
 import okhttp3.OkHttpClient
-import javax.inject.Inject
 
 @Module
 class RepositoryModule {
-
-    @Inject
-    lateinit var dateUtils: DateUtils
 
     @Provides
     fun provideConversationsRepository(ncApi: NcApi, userProvider: CurrentUserProviderNew): ConversationsRepository {
@@ -62,7 +58,7 @@ class RepositoryModule {
     }
 
     @Provides
-    fun provideSharedItemsRepository(ncApi: NcApi): SharedItemsRepository {
+    fun provideSharedItemsRepository(ncApi: NcApi, dateUtils: DateUtils): SharedItemsRepository {
         return SharedItemsRepositoryImpl(ncApi, dateUtils)
     }
 

